Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h

```diff
@@ -825,6 +827,19 @@ namespace internal {
 #ifdef CGAL_PMP_REMESHING_VERBOSE
       std::cout << "Equalize valences..." << std::endl;
 #endif
+
+      typedef typename boost::property_map<PM, CGAL::dynamic_vertex_property_t<int> >::type Vertex_degree;
+      Vertex_degree degree = get(CGAL::dynamic_vertex_property_t<int>(), mesh_);
+
+      for(vertex_descriptor v : vertices(mesh_)){
+        put(degree,v,0);
+      }
+      for(halfedge_descriptor h : halfedges(mesh_))
+      {
+        vertex_descriptor t = target(h, mesh_);
+        put(degree, t, get(degree,t)+1);
```

Comment:
@janetournois shall the degree depend only on the selected edges? This matters only for vertices on the boundary of a selection.

Reply:
The degree of a vertex that stands on the boundary of the selection (i.e. on a `PATCH_BORDER`) should count only internal edges.
Here all edges are counted, right?

Reply:
Note that in your code, IRC you were using `degree()` so Andreas' code does the same thing the original (to be checked)

Reply:
But in the original code we call the `degree()` function for the vertex which does not distinguish either.

Reply:
You are right.
`valence(v)` and `target_valence(v)` (which is 4 on borders and 6 elsewhere) both take into account only the "real" borders, not the patch borders.

It's not wrong, but maybe we can improve that, when a vertex stands at the limit between a patch to be remeshed and a patched to be left as is.
We can keep it this way for now.